### PR TITLE
adds swagger

### PR DESCRIPTION
**Pull request by arifratul41**

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,7 +1,7 @@
 spring.datasource.url=jdbc:mysql://localhost:3306/studentlog
 spring.datasource.driverClassName=com.mysql.cj.jdbc.Driver
 spring.datasource.username=root
-spring.datasource.password=root1234
+spring.datasource.password=thisissparta
```
Comment (ishmum123) [inline src/main/resources/application.properties]: ?

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -11,3 +11,8 @@ spring.jpa.properties.hibernate.format_sql=true
 spring.autoconfigure.exclude[0]=org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration
 
 #spring.liquibase.drop-first=true
+server.port=9000
```
Comment (ishmum123) [inline src/main/resources/application.properties]: ?

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -11,3 +11,8 @@ spring.jpa.properties.hibernate.format_sql=true
...
+
+# swagger-ui custom path
+springdoc.api-docs.enabled=true
+springdoc.swagger-ui.path=/swagger-ui.html
```
Comment (ishmum123) [inline src/main/resources/application.properties]: is this not the default path?

File: src/main/java/com/ideal/studentlog/helpers/dtos/TestDTO.java [hunk 4]
```diff
@@ -11,12 +12,13 @@
 public class TestDTO {
 
     @NonNull
-    String subject;
+    @Schema(description = "Subject", example = "Bengali")
```
Comment (ishmum123) [inline src/main/java/com/ideal/studentlog/helpers/dtos/TestDTO.java]: Please create the branch from the latest main to avoid conflicts

**Review by ishmum123 — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 5]
```diff
@@ -119,6 +119,11 @@
 <!--			<version>RELEASE</version>-->
 			<scope>compile</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.springdoc</groupId>
+			<artifactId>springdoc-openapi-ui</artifactId>
+			<version>1.5.8</version>
```
Comment (ishmum123) [inline pom.xml]: Please use variables for versioning

**Review by ishmum123 — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 1 above)
Comment (arifratul41) [reply src/main/resources/application.properties]: this is my localhost password. I think we should add this application.properties file to .gitignore

File: src/main/resources/application.properties (hunk 2 above)
Comment (arifratul41) [reply src/main/resources/application.properties]: Same. Application.properties should be in .gitignore

File: src/main/resources/application.properties (hunk 3 above)
Comment (arifratul41) [reply src/main/resources/application.properties]: yes. but added there explicitly still to address that can create custom swagger ui URL

File: src/main/java/com/ideal/studentlog/helpers/dtos/TestDTO.java (hunk 4 above)
Comment (arifratul41) [reply src/main/java/com/ideal/studentlog/helpers/dtos/TestDTO.java]: Done

File: pom.xml (hunk 5 above)
Comment (arifratul41) [reply pom.xml]: https://github.com/ishmum123/studentlog/pull/28/commits/d774b39665d7d33cb5d17ac01b3c2793f1e5d969

File: src/main/resources/application.properties (hunk 1 above)
Comment (ishmum123) [reply src/main/resources/application.properties]: okay, but why add to this PR then?

File: src/main/resources/application.properties (hunk 3 above)
Comment (ishmum123) [reply src/main/resources/application.properties]: I can't foresee a case where we would need to change swagger's URL

**Pull request merged**